src/main/java/org/opensearch/agent/tools/CreateAlertTool.java

```diff
@@ -62,29 +63,53 @@ public class CreateAlertTool implements Tool {
     private String description = DEFAULT_DESCRIPTION;
 
     private final Client client;
+    @Getter
     private final String modelId;
-    private final String TOOL_PROMPT_TEMPLATE;
+    @Getter
+    private final String toolPrompt;
 
     private static final String MODEL_ID = "model_id";
     private static final String PROMPT_FILE_PATH = "CreateAlertDefaultPrompt.json";
     private static final String DEFAULT_QUESTION = "Create an alert as your recommendation based on the context";
     private static final Map<String, String> promptDict = ToolHelper.loadDefaultPromptDictFromFile(CreateAlertTool.class, PROMPT_FILE_PATH);
 
-    public CreateAlertTool(Client client, String modelId, String modelType) {
+    public enum ModelType {
+        CLAUDE,
+        OPENAI;
+
+        public static ModelType from(String value) {
+            if (value.isEmpty()) {
+                return ModelType.CLAUDE;
+            }
+            try {
+                return ModelType.valueOf(value.toUpperCase(Locale.ROOT));
+            } catch (Exception e) {
+                log.error("Wrong Model type, should be CLAUDE or OPENAI");
+                return ModelType.CLAUDE;
+            }
+        }
+    }
```

Comment:
Wait a bit. The code is the same as the ModelType in PPL tool. Can you extract it outside the function for code reusage?

Reply:
They are not exactly the same,  this tool doesn't support `FINETUNE` model. If we want to reuse this method, we'd better refactor it as well and I think it should also applies to other tools like `CreateAnomalyDetectorTool`.

So I don't think it's a blocker to this PR.

Reply:
I see. Then the code is good.